### PR TITLE
WIP #7468 Move debounce from ZStream to ZPipeline

### PR DESCRIPTION
**Pull request by tewecske**
I need help with this. The runScoped call doesn't compile.

### CODE REVIEW

**Comment by adamgfraser** [comment]
@tewecske Thanks for your work on this! Will take a look!

File: streams/shared/src/main/scala/zio/stream/ZPipeline.scala [hunk 1]
```diff
@@ -228,6 +228,104 @@ object ZPipeline extends ZPipelinePlatformSpecificConstructors {
...
+                    )
+              }
+            )
+
+          ZChannel.scoped[Env](producer.runScoped.forkScoped) *>
+            new ZPipeline(consumer(NotStarted))
+        }
+      }
+    )
+  }
```
Comment (paulpdaniels) [inline streams/shared/src/main/scala/zio/stream/ZPipeline.scala:327]: I haven't checked it locally but I would suggest running everything in `ZChannel` and only then wrapping it in a `ZPipeline`

i.e.

```scala
def debounce[Env, Err, In](d: => Duration)(implicit trace: Trace): ZPipeline[Env, Err, In, In] = {
    import ZStream.DebounceState
    import ZStream.DebounceState._
    import ZStream.HandoffSignal
    import ZStream.HandoffSignal._

new ZPipeline(ZChannel.unwrap {
   // ...  

  ZChannel.unwrapScoped(producer.runScoped.forkScoped as consumer(NotStarted))
})

```

**Comment by jdegoes** [comment]
@tewecske This is not compiling. Please re-open if you'd like to work on it more, and perhaps one of us can pair with you on Zoom to get this over the hump!

**Comment by tewecske** [comment]
@jdegoes  I can't reopen this but I tried to fix it. @adamgfraser helped and there is a new commit with my attempts but still failing so I need some help. I also rebased it to series/2,x

**Comment by adamgfraser** [comment]
Will take a look this weekend.

**Pull request closed without merging**